protobuf/lib/src/protobuf/pb_map.dart

```diff
@@ -39,6 +39,35 @@ class PbMap<K, V> extends MapBase<K, V> {
     _wrappedMap[key] = value;
   }
 
+  @override
+  bool operator ==(other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other is! PbMap) {
+      return false;
+    }
+    if (other.length != length) {
+      return false;
+    }
+    if (other.hashCode != hashCode) {
+      return false;
+    }
+    for (final key in keys) {
+      if (other[key] != this[key]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  int get hashCode {
+    return hashObjects(_wrappedMap.keys
+        .map((key) => hash2(key.hashCode, _wrappedMap[key].hashCode))
+        .toList(growable: false)
+          ..sort());
```

Comment:
I think we can just rely on the default dart map being ordered (it is a LinkedHashMap)?

Reply:
No of course not! - different order of insertion should still result in the same hashcode. I think we can xor the hashcodes of all the entries.

Reply:
hash2 takes two Objects, and calls their hashCodes.
So the calls the hashCode here are unnecessary.

hashObjects takes an Iterable of Objects and calls the hashCodes.

In effect we are doing a lot of x.hashCode.hashCode.
At best that seems inefficient, and may result in a poor hashCode if the underlying combiner doesn't have good dispersal.
At least we can fix the two calls here.